test/built-ins/Array/prototype/sort/comparefn-nonfunction-call-throws.js

```diff
@@ -15,40 +15,81 @@ info: >
 features: [Symbol]
 ---*/
 
-var sample = [1, 2, 3];
+var sample = [1];
+var poisoned = {
+  get length() {
+    throw new Test262Error("IsCallable(comparefn) should be observed before this.length");
+  }
+};
 
 assert.throws(TypeError, function() {
  sample.sort(null);
-});
+}, "sample.sort(null);");
+
+assert.throws(TypeError, function() {
+  [].sort.call(poisoned, null);
+}, "[].sort.call(poisoned, null);");
```

Comment:
I kept both tests using a regular array with one entry `[1]` and another one with an object containing a poisoned `length` property.